src/ol/proj/proj.js

```diff
@@ -370,22 +391,124 @@ ol.proj.Projection.prototype.getPointResolution_ = function(resolution, point) {
 
 
 /**
- * Get the resolution of the point in degrees or distance units.
- * For projections with degrees as the unit this will simply return the
- * provided resolution. The default for other projections is to estimate
- * the point resolution by transforming the 'point' pixel to EPSG:4326,
- * measuring its width and height on the normal sphere,
- * and taking the average of the width and height.
- * An alternative implementation may be given when constructing a
- * projection. For many local projections,
- * such a custom function will return the resolution unchanged.
- * @param {number} resolution Resolution in projection units.
- * @param {ol.Coordinate} point Point.
- * @return {number} Point resolution in projection units.
+ * Calculate the length of a line string in projection units except
+ * if the projection's units are DEGREES, when a length in meters is returned.
```

Comment:
True about pixels. For everything else, I think the return value should be in meters.